IdentityCore/src/oauth2/MSIDOauth2Factory.m

```diff
@@ -186,7 +187,8 @@ - (BOOL)fillBaseToken:(MSIDBaseToken *)token
     token.authority = configuration.authority;
     token.clientId = configuration.clientId;
     token.additionalServerInfo = response.additionalServerInfo;
-    token.homeAccountId = response.idTokenObj.userId;
+    token.accountIdentifier = [[MSIDAccountIdentifier alloc] initWithLegacyAccountId:response.idTokenObj.userId
+                                                                       homeAccountId:response.idTokenObj.userId];
```

Comment:
Just curious, why not `token.accountIdentifier.homeAccountId` here? :)

Reply:
There's no home account id (client info) for base Oauth 2 case. Because we still need some kind of identifier for MSAL for other Oauth2 providers, we use something else from the id token here :) In AAD factory we override it with whatever comes from the client info...